File: Contest2/Orange/insert-a-node-at-a-specific-position-in-a-linked-list.py
def insertNodeAtPosition(head, data, position):
    node = SinglyLinkedListNode(data)
    curr = head
    # print (node.data)
    if position==0:
        node.next = head
    else:
        position-=1
        while(position):
            curr=curr.next
            position-=1
        node.next=curr.next
        curr.next=node
        
        
    return head

File: Contest2/Orange/insert-a-node-at-a-specific-position-in-a-linked-list.cpp

/*
 * For your reference:
 *
 * SinglyLinkedListNode {
 *     int data;
 *     SinglyLinkedListNode* next;
 * };
 *
 */
SinglyLinkedListNode* insertNodeAtPosition(SinglyLinkedListNode* head, int data, int position)
{
    SinglyLinkedListNode* temp= new SinglyLinkedListNode(data);
    temp->data=data;
    temp->next=NULL;
    if(position==1)
    {
        temp->next=head;
        head=temp;
        return head;
    }
    SinglyLinkedListNode* temp1=head;
    for(int i=0;i<position-1;i++)
        temp1=temp1->next;
    temp->next=temp1->next;
    temp1->next=temp;
    return head;
}
